packages/integration-karma/test/shadow-dom/Element-properties/Element.id.spec.js

```diff
@@ -9,3 +11,55 @@ it('should allow setting id properties manually', () => {
     shadowRoot.querySelector('div').id = 'something';
     expect(shadowRoot.querySelector('div').getAttribute('id')).toBe('something');
 });
+
+describe('scoped-ids', () => {
+    describe('expressions', () => {
+        describe('custom elements', () => {
+            it('should render expected id attribute value when its value is set to `undefined`', () => {
+                const elm = createElement('x-foo', { is: IdValueUndefined });
+                expect(() => {
+                    document.body.appendChild(elm);
+                }).toLogErrorDev(
+                    /\[LWC error\]: Invalid id value "undefined". The id attribute must contain a non-empty string./
+                );
+                const child = elm.shadowRoot.querySelector('x-child');
+                expect(child.getAttribute('id')).toEqual(null);
```

Comment:
Looks like this skipped test was incomplete. `x-child` doesn't have an `id` attribute so we need to add it. I tested it and saw that the value should be `'undefined'`.

Reply:
Updated the test. It seems like there is a difference in behavior for custom elements v/s standard html elements. When id value is undefined, standard html element return `null`.

Reply:
Yeah, it has to do with how we set content vs IDL attribute for native vs custom elements.

Reply:
Logged #1769 to document the difference. Added a comment in the test to link to the issue.